rust-src/curve_arithmetic/src/curve_arithmetic.rs

```diff
@@ -1,24 +1,20 @@
+use anyhow::Result;
```

Comment:
I don't think we should import Result like this since it is ambiguous with the std::Result.

Either rename or use qualified.